features/support/mock_mailgun.rb

```diff
@@ -0,0 +1,27 @@
+require 'webrick'
+
+Before('@mock_mailgun') do
+  @mailgun_server = WEBrick::HTTPServer.new(
+    Port: 9293,
+    StartCallback: -> { @mailgun_server_running = true },
+    Logger: Rails.logger,
+    AccessLog: Rails.logger
+  )
+
+  @mailgun_thread = Thread.new { @mailgun_server.start }
+  Timeout.timeout(1) { :wait until @mailgun_server_running }
+end
+
+After('@mock_mailgun') do
+  @mailgun_server.shutdown
```

Comment:
If I stopped cucumber early by `^C` this would raise and the interface/port would be forever bound. We'll have to keep an eye on this.